software-architecture/user-stories.qmd

```diff
@@ -0,0 +1,73 @@
+---
+title: "User stories"
+date: last-modified
+---
+
+[User stories](https://en.wikipedia.org/wiki/User_story) are a natural language description of specific or desired features of a software product. User stories are often structured in the form of:
+
+> As a USER, I want to ACTION because REASON.
+
+Here we list the user stories for each of our anticipated / target users.
+
+## User Access Administrator
+
+|As a... |I want to... |Because... |
+| --- | --- | --- |
+|User Access Administrator|set up user accounts|so that people can work in the system|
+|User Access Administrator|assign a user to a role/group|so that they will receive the correct permissions to work in the system|
+|User Access Administrator|see the individuals assigned to each role|so that I can easily audit who has access to which parts of the system|
+|User Access Administrator|see the groups/roles assigned to an individual|so that I can easily audit what kind of access a user has/has had|
+
+## Data Request Admin
+
+|As a... |I want to... |Because... |
+| --- | --- | --- |
+|Data Request Admin|see the applications submitted|so that I can check if they are completed correctly|
+|Data Request Admin|send an application back to the requester|so that they can fill out any missing details|
+|Data Request Admin|comment on a data request|so that I can inform either the requester or the data controller of my feedback|
+|Data Request Admin|forward a request to the data controller|so that they can make a decision on whether or not to grant access to the data|
+|Data Request Admin|put together a data extract script|so that I can provide the data requested|
+|Data Request Admin|pull the data out of the system|so that I can make the data available to the requestor|
+|Data Request Admin|see how many rows of data a specific request will generate|so that I can determine if the tables have been linked correctly|
+|Data Request Admin|select between formats for download of data|so that I can send the most appropriate format to the data requester|
```

Comment:
These "stories" shouldn't need to be done by the Data Request Admin, these should all be done automatically after a project proposal is approved.

Reply:
@lwjohnst86 Although you still need to document them as a user story so that we have an argument for setting the system up to do it automatically...